server/tests/test.js

```diff
@@ -0,0 +1,130 @@
+import 'babel-polyfill'
+import { product, adminData, userData } from '../tests/constants'
+
+const { MongoMemoryServer } = require('mongodb-memory-server')
+const prepareApp = require('../app')
+const request = require('supertest')
+const mongoose = require('mongoose')
+const Cart = require('../models/Cart')
+const Product = require('../models/Product')
+
+describe('Test', function () {
+  let mongoServer, app
+
+  beforeAll(async () => {
+    mongoServer = new MongoMemoryServer()
+    const mongoUri = await mongoServer.getUri()
+
+    const server = await prepareApp(mongoUri)
```

Comment:
@AStashevskaya `prepareApp` is not async function, but you can specify callback in `mongoose.connect` to resolve promise when app connected to `mongo` server